.gitlab-ci.yml

```diff
@@ -2,350 +2,165 @@
 # paritytech/scripts
 #
 
-
 stages:
   - build
   - deploy
   - destroy
 
-
-
-
-
-
 variables:
-  # CONTAINER_IMAGE: registry.gitlab.com/$CI_PROJECT_PATH
-  # DOCKER_HOST: tcp://docker:2375
-  DOCKER_HOST: tcp://localhost:2375
-  DOCKER_DRIVER: overlay2
-
+  # CONTAINER_IMAGE:    registry.gitlab.com/$CI_PROJECT_PATH
+  # DOCKER_HOST:        tcp://docker:2375
+  DOCKER_HOST:          tcp://localhost:2375
+  DOCKER_DRIVER:        overlay2
+  CONTAINER_IMAGE:      $CI_JOB_NAME
+  CONTAINER_TAG:        latest
+  DOCKERFILE_DIR:       $CI_JOB_NAME
 
 
-.kubernetes-build:                 &kubernetes-build
+.build:                 &docker_build
+  stage:                build
+  image:                docker:stable
+  variables:
+    REGISTRY_PATH:      paritytech
   only:
+    variables:
+      - $DOCKERIMAGE == $CI_JOB_NAME
     refs:
       - schedules
       - web
-    kubernetes: active
+    kubernetes:         active
   tags:
     - kubernetes-parity-build
-
-
-.build:                            &docker_build
-  stage: build
-  <<:                              *kubernetes-build
-  image:                           docker:stable
   services:
     - docker:dind
   before_script:
     - test -z "$DOCKERIMAGE" && echo "DOCKERIMAGE must be defined" && exit 1
-    - docker login -u "${Docker_Hub_User_Parity}" -p "${Docker_Hub_Pass_Parity}"
-    - docker info
-  script:
-    # - docker pull $CONTAINER_IMAGE:$CONTAINER_TAG || true
-    # - docker build --cache-from $CONTAINER_IMAGE:$CONTAINER_TAG --tag $CONTAINER_IMAGE:$CI_BUILD_REF --tag $CONTAINER_IMAGE:$CONTAINER_TAG .
+  script:               &build_and_push_image
     - cd dockerfiles/
     - export CONTAINER_DATE_TAG="${CI_COMMIT_SHORT_SHA}-$(date +%Y%m%d)"
     - docker build --no-cache
         --build-arg VCS_REF="${CI_COMMIT_SHA}"
         --build-arg BUILD_DATE="$(date +%Y%m%d)"
-        --tag parity/$CONTAINER_IMAGE:$CONTAINER_DATE_TAG
-        --tag parity/$CONTAINER_IMAGE:$CONTAINER_TAG
+        --build-arg REGISTRY_PATH=$REGISTRY_PATH
+        --tag $REGISTRY_PATH/$CONTAINER_IMAGE:$CONTAINER_DATE_TAG
+        --tag $REGISTRY_PATH/$CONTAINER_IMAGE:$CONTAINER_TAG
         --file $DOCKERFILE_DIR/Dockerfile .
-    - docker push parity/$CONTAINER_IMAGE:$CONTAINER_DATE_TAG
-    - docker push parity/$CONTAINER_IMAGE:$CONTAINER_TAG
-  after_script:
+    # Push to Dockerhub
+    - docker login -u "${Docker_Hub_User_Parity}" -p "${Docker_Hub_Pass_Parity}"
+    - docker info
+    - docker push $REGISTRY_PATH/$CONTAINER_IMAGE:$CONTAINER_DATE_TAG
+    - docker push $REGISTRY_PATH/$CONTAINER_IMAGE:$CONTAINER_TAG
     - docker logout
-  environment:
-    name: parity-build
-
-.build_to_own_reg:                 &docker_build_to_own_reg
-  <<:                              *docker_build
-  before_script:
-    - test -z "$DOCKERIMAGE" && echo "DOCKERIMAGE must be defined" && exit 1
+    # Push to CI registry
```

Comment:
Now it's a way better, you build once, but actually it's now not clear why every image that uses this temlate gets pushed to both registries. 
I thought it would be better to separately anchor what you have under `# Push to Dockerhub` like
```
- docker login -u "${Docker_Hub_User_Parity}" -p "${Docker_Hub_Pass_Parity}";
  docker info;
  docker push $REGISTRY_PATH/$CONTAINER_IMAGE:$CONTAINER_DATE_TAG;
  docker push $REGISTRY_PATH/$CONTAINER_IMAGE:$CONTAINER_TAG; 
  docker logout;                                           &push_to_dockerhub
```
and then use this as
```
script:
  - some previous steps
  - *push_to_dockerhub
```
and do the same for `push_to_gitlab`